ui/etc/createFragmentTypes.js

```diff
@@ -0,0 +1,37 @@
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+fetch(process.env.REACT_APP_GRAPHQL_ENDPOINT, {
+  method: 'POST',
+  headers: { 'Content-Type': 'application/json' },
+  body: JSON.stringify({
+    variables: {},
+    query: `
+      {
+        __schema {
+          types {
+            kind
+            name
+            possibleTypes {
+              name
+            }
+          }
+        }
+      }
+    `,
+  }),
+})
+  .then(result => result.json())
+  .then((result) => {
+    // here we're filtering out any type information unrelated to unions or interfaces
+    const filteredData = result.data.__schema.types.filter(type => type.possibleTypes !== null);
+    // eslint-disable-next-line
+    result.data.__schema.types = filteredData;
+    fs.writeFile('./src/services/apolloClient/fragmentTypes.json', JSON.stringify(result.data), (err) => {
+      if (err) {
+        console.error('Error writing fragmentTypes file', err);
+      } else {
+        console.log('Fragment types successfully extracted!');
+      }
+    });
+  });
```

Comment:
Hm, now we're running the script on npm start and generating a file that goes in our src and gets checked in. This feels pretty weird to me, but I'm not sure I have a better suggestion. It would be nice if the script ran at build time and created an artifact in the build that never had to be checked in, but making that work with webpack feels like an ordeal (unless we have someone on our team who actually feels like they understand webpack 😛). 

Either way the file should have a newline at the end, and I suggest moving it to the scripts/ directory instead of etc/. Maybe also create a github issue (low prio, but still) to come up with a more elegant solution for this. What do you think?

Reply:
PS. Nice work figuring out a solution though. Feels like it might have been a tricky one 😄

Reply:
Agreed, would be nice to not have this generated file laying around 🙂.

Added a newline and moved it to `scripts/` at least. I'll create an issue for finding a better solution.